src/testsrc/java/com/epam/jamp2/service/TestCalculationCommandExecutionService.java

```diff
@@ -0,0 +1,133 @@
+        this.expected = expected;
+    }
+
+    private static final Map<String, BigDecimal> fxRates = new HashMap<>();
+    static {
+        fxRates.put("gbp => cny", BigDecimal.valueOf(10));
+        fxRates.put("cny => gbp", BigDecimal.valueOf(0.1));
+        fxRates.put("cny => cny", BigDecimal.valueOf(1));
+        fxRates.put("gbp => gbp", BigDecimal.valueOf(1));
+    }
+
+    @Before
+    public void beforeTest() throws IOException, UnknownCurrencyException {
+        this.target = new CalculationCommandExecutionServiceImpl();
+
+        try {
+            fxRatesService = mock(FxRatesService.class);
+
+            // Mocks the possible calls to the FxRatesService.
+            BigDecimal o = BigDecimal.valueOf(0.0);
+            Optional<BigDecimal> leftToResultRate = Optional.ofNullable(
+                    fxRates.get(command.getLeftOperand().getCurrencyCode().orElse("null") + " => "
+                            + command.getResultCurrencyCode().orElse("null")));
+            Optional<BigDecimal> rightToResultRate = Optional.ofNullable(
+                    fxRates.get(command.getRightOperand().getCurrencyCode().orElse("null") + " => "
+                            + command.getResultCurrencyCode().orElse("null")));
+            BigDecimal leftInTarget = command.getLeftOperand().getValue().multiply(leftToResultRate.orElse(o));
+            BigDecimal rightInTarget = command.getRightOperand().getValue().multiply(rightToResultRate.orElse(o));
+            OngoingStubbing leftCall = when(fxRatesService.convert(command.getLeftOperand().getCurrencyCode().get()
+                    , command.getResultCurrencyCode().get()
+                    , command.getLeftOperand().getValue()));
+            leftCall.thenReturn(leftInTarget);
+            OngoingStubbing rightCall = when(fxRatesService.convert(command.getRightOperand().getCurrencyCode().get()
+                    , command.getResultCurrencyCode().get()
+                    , command.getRightOperand().getValue()));
+            rightCall.thenReturn(rightInTarget);
+
+            this.target.setFxRatesService(fxRatesService);
+        } catch (java.util.NoSuchElementException e) {
+            throw new java.lang.UnsupportedOperationException("My mock it still not smart enough to deal with " +
+                    "null values for currencies.");
+        }
+    }
+
+    @Test
+    public void test_getConvertedValue() throws IOException, UnknownCurrencyException {
+
+        BigDecimal leftInTarget = target.getConvertedValue(command.getLeftOperand(), command.getResultCurrencyCode());
+        BigDecimal rightInTarget = target.getConvertedValue(command.getRightOperand(), command.getResultCurrencyCode());
+        BigDecimal errorTolerance = expected.getValue().multiply(BigDecimal.valueOf(0.000001)); // i.e. ±0.00001% for float precision
+
+        if(command.getOperation().equals(Operation.ADD)) {
```

Comment:
The need to have such a hook in parametrized test is another signal that test is not universal. It's better to split into two different tests in such cases. But in this particular case I would say if you tested ADD operation it doesn't make much sense to separately test SUBSTRUCT since A + B === A + (-B)